Dockerfile

```diff
@@ -0,0 +1,37 @@
+# Build stage
+FROM python:3.8 AS builder
+
+ENV PYTHONUNBUFFERED=1
+
+# Set the working directory inside the container
+WORKDIR /app
+
+# Copy the requirements file first to leverage Docker cache
+COPY requirements.txt /app/
+
+# Install the Python dependencies
+RUN pip install --no-cache-dir -r requirements.txt
+
+# Copy the rest of the application code into the container
+COPY . /app
```

Comment:
You can use one `COPY` to copy one file once not to duplicate files and have less layer 
```suggestion
COPY . /app/

# Install the Python dependencies
RUN pip install --no-cache-dir -r requirements.txt
```